### PR TITLE
[IMPROVEMENT] Don't display emojis inside code blocks

### LINKED ISSUE #1185: [IMPROVEMENT] Emoji substitution on text should not happen inside of a code block

### PR DESCRIPTION
**Pull request by cardoso**
@RocketChat/ios

Closes #1185

### CODE REVIEW

**Comment by rafaelks** [comment]
Conflicts 🙈

File: Rocket.Chat/External/RCEmojiKit/NSAttributedString+CustomEmojis.swift [hunk 1]
```diff
@@ -24,7 +24,16 @@ extension NSAttributedString {
...
 
-            for match in matches {
+            // exclude matches inside code tags
+            let codeRanges = attributedString.string.codeRanges()
+
+            let filteredMatches = matches.filter { match in
+                !codeRanges.contains { range in
+                    match.range(at: 0).location >= range.location && match.range(at: 0).length <= range.length
+                }
+            }
```
Comment (rafaelks) [inline Rocket.Chat/External/RCEmojiKit/NSAttributedString+CustomEmojis.swift]: Looks like we're duplicating code? Is there any reason for that @cardoso?

**Comment by cardoso** [comment]
Made the changes @rafaelks

**Comment by rafaelks** [comment]
@cardoso Found one bug, if you type an emoji after a code block, it's not transforming the emoji. Example:

> ```
> :beer:
> ```
> 
> :beer:

**Comment by cardoso** [comment]
Fixed the bug @rafaelks

**Comment by rafaelks** [comment]
@cardoso Just tried your branch:

simulator screen shot - iphone se - 2018-02-26 at 18 05 40

**Comment by cardoso** [comment]
Can you take a look again @rafaelks ? I added unit tests this time.

**Comment by rafaelks** [comment]
Perfect @cardoso!!! 👏

**Pull request merged**